cpp/src/arrow/result.h

```diff
@@ -215,6 +215,9 @@ class Result {
     return *this;
   }
 
+  /// Compare to another Result.
+  bool operator==(const Result& other) const { return variant_ == other.variant_; }
```

Comment:
Does C++ create the corresponding `operator!=` automatically?

Reply:
No, I'll add it